### PR TITLE
fix: incorrect MLS verification system messages WPB-6909

### PR DESCRIPTION
**Pull request by typfel**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-6909" title="WPB-6909" target="_blank">BugWPB-6909</a>  [iOS] Fingerprints verified system message appearing in E2EI MLS conversation
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

There are several issues with MLS verification:

1. Conversation always had a default `mlsVerificationStatus` which messes up the state transitions since we don't want insert any system message for the initial verification state.
2. When MLS groups were created with two commits: first empty group with self user and then other participants were added. This causes problems with MLS verification since after the first commit all conversations will be considered verified and then potentially we'll transition to degraded when we add the participants.
3. The MLS verification system message cells weren't implemented.

### Causes

1. CoreData doesn't support optional scalar values, in our case Int16 will default to 0 which was mapped to `.verified`.
2. -
3. -

### Solutions

1. Map 0 to nil by changing enum start numbering to 1
2. Initialise MLS group with a single commit
3. Add missing cell implementation

### Attachments (Optional)

Simulator Screenshot - iPhone 15 Pro - 2024-02-29 at 23 41 25

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: wire-ios-data-model/Source/MLS/MLSVerificationStatus.swift [hunk 1]
```diff
@@ -20,8 +20,8 @@ import Foundation
 
 public enum MLSVerificationStatus: Int16 {
 
+    case notVerified = 1
```
Comment (netbe) [inline wire-ios-data-model/Source/MLS/MLSVerificationStatus.swift:26]: add a comment here for the reason of starting here

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/Components/ConversationVerifiedSystemMessageSectionDescription.swift [hunk 2]
```diff
@@ -42,13 +42,14 @@ final class ConversationVerifiedSystemMessageSectionDescription: ConversationMes
     let accessibilityLabel: String?
 
     init() {
-        let title = NSAttributedString(
-            string: L10n.Localizable.Content.System.isVerified,
-            attributes: [.font: UIFont.mediumFont, .foregroundColor: LabelColors.textDefault]
-        )
+        let title = NSAttributedString.markdown(from: L10n.Localizable.Content.System.Mls.conversationIsVerified(URL.wr_e2eiLearnMore), style: .systemMessage)
+//        let title = NSAttributedString(
```
Comment (KaterinaWire) [inline wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/Components/ConversationVerifiedSystemMessageSectionDescription.swift]: don't forget to remove the commented code

File: wire-ios-data-model/Source/MLS/MLSService.swift [hunk 3]
```diff
@@ -34,6 +34,8 @@ public protocol MLSServiceInterface: MLSEncryptionServiceInterface, MLSDecryptio
 
     func createGroup(for groupID: MLSGroupID, with users: [MLSUser]) async throws
 
+    func createGroup(for groupID: MLSGroupID, parentGroupID: MLSGroupID?) async throws
```
Comment (KaterinaWire) [inline wire-ios-data-model/Source/MLS/MLSService.swift:37]: Is it part of this PR?

File: wire-ios-data-model/Source/MLS/MLSService.swift [hunk 4]
```diff
@@ -453,7 +455,7 @@ public final class MLSService: MLSServiceInterface {
         try await addMembersToConversation(with: usersWithSelfUser, for: groupID)
     }
 
-    func createGroup(
+    public func createGroup(
```
Comment (KaterinaWire) [inline wire-ios-data-model/Source/MLS/MLSService.swift:458]: same here

**Review by KaterinaWire — APPROVED** [review]

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/Components/ConversationVerifiedSystemMessageSectionDescription.swift (hunk 2 above)
Comment (typfel) [reply wire-ios/Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/ConfigurationMessageCell/Components/ConversationVerifiedSystemMessageSectionDescription.swift]: 💥

**Review by netbe — COMMENTED** [review]
LGTM;) just added comment

**Review by David-Henner — APPROVED** [review]

File: wire-ios-data-model/Source/MLS/MLSVerificationStatus.swift (hunk 1 above)
Comment (typfel) [reply wire-ios-data-model/Source/MLS/MLSVerificationStatus.swift:26]: Done

**Pull request merged**